src/bin/fish.rs

```diff
@@ -261,7 +351,57 @@
 
 /// Parse init files. exec_path is the path of fish executable as determined by argv[0].
 fn read_init(parser: &Parser, paths: &ConfigPaths) {
-    let datapath = str2wcstring(paths.data.as_os_str().as_bytes());
+    let datapath = if cfg!(feature = "installable") {
+        let (_, Some(home)) = fish::env::get_user_home() else {
+            FLOG!(
+                error,
+                "Cannot find home directory and will refuse to read configuration"
+            );
+            return;
+        };
+
+        let p = PathBuf::from(home).join(paths.data.clone());
+        let Some(homestr) = p.to_str() else {
+            FLOG!(
+                error,
+                "Cannot decode home directory and will refuse to read configuration"
+            );
+            return;
+        };
+        str2wcstring(homestr.as_bytes())
+    } else {
+        str2wcstring(paths.data.as_os_str().as_bytes())
+    };
+
+    #[cfg(feature = "installable")]
+    {
+        // When fish is installable, we write the version to a file,
+        // now we check it.
+        let verfile =
+            PathBuf::from(fish::common::wcs2osstring(&datapath)).join("fish-install-version");
+        let Ok(version) = std::fs::read_to_string(verfile) else {
+            let escaped_pathname = escape(&datapath);
+            FLOGF!(
+                error,
+                "Fish cannot find its asset files in '%ls'.\n\
+                 Refusing to read configuration because of this.",
```

Comment:
should probably also log the underlying error, like

```rust
let version = match read_to-string(verfile) {
	Ok(version) = version,
	Err(err) => {
		FLOG!(error, "fish cannot read version file from installed asset files:", err);
	}
};
```

Reply:
also this disallows running `target/debug/fish`. We should probably skip the version check if we're colocated like that,
because `share/` has the right version by definition

Reply:
when running directly as `target/*/fish`, we should use the asset files in the source tree but we currently can't build them without cmake